src/einsteinpy/coordinates/utils.py

```diff
@@ -6,193 +6,193 @@
 _c = constant.c.value
 
 
-def cartesian_to_spherical_fast(
-    t, x, y, z, v_x=None, v_y=None, v_z=None, velocities_provided=False
-):
-    if velocities_provided:
-        return cartesian_to_spherical(t, x, y, z, v_x, v_y, v_z)
-    return cartesian_to_spherical_novel(t, x, y, z)
+@jit
+def calculate_coordinates(r, th, p, alpha):
```

Comment:
Is the method in https://github.com/einsteinpy/einsteinpy/pull/644#discussion_r1343410510 causing any issues ? (Apart from `not any`) ? That way should lead to clearer and more compact code. The idea is that the position and velocity conversions are `jit`-compiled, and these function names (`cartesian_to_spherical` vs `cartesian_to_spherical_novel`) are changed to something more comprehensible.

Reply:
Alright thanks for the pointers, will refactor based on that approach